SU2_CFD/include/solvers/CFVMFlowSolverBase.inl

```diff
@@ -2581,16 +2582,36 @@ void CFVMFlowSolverBase<V, FlowRegime>::Friction_Forces(const CGeometry* geometr
 
       } else {
 
+        unsigned short iSpecies, nSpecies = config->GetnSpecies();
+
         const auto thermal_conductivity_tr = nodes->GetThermalConductivity(iPoint);
         const auto thermal_conductivity_ve = nodes->GetThermalConductivity_ve(iPoint);
         const auto Grad_PrimVar            = nodes->GetGradient_Primitive(iPoint);
+        const auto PrimVar                 = nodes->GetPrimitive(iPoint);
+        const auto Ds                      = nodes->GetDiffusionCoeff(iPoint);
+        const auto hs                      = nodes->GetEnthalpys(iPoint);
+
+        su2double rho                      = PrimVar[RHO_INDEX]; 
+        su2double dYsn;
+
+        /*--- Compute enthalpy transport to surface due to mass diffusion ---*/ 
+        su2double sumJhs = 0.0;
+        for (iDim = 0; iDim < nDim; iDim++) {
+          for (iSpecies = 0; iSpecies < nSpecies; iSpecies++) {
+            dYsn = Grad_PrimVar[RHOS_INDEX+iSpecies][iDim]/rho*UnitNormal[iDim];
+            sumJhs += rho*Ds[iSpecies]*dYsn*hs[iSpecies];
+          }
+        }
```

Comment:
Clean this up please, you're dividing by rho, then multiplying on the line below.
And use the geometry toolbox for dot products with the unit normal, saves the loop over nDim (which is in the wrong order, should be inside the species loop).

Reply:
Ok, I'll do the same for the compressible and incompressible heat fluxes as well. It looks like there are a lot of iDim loops that can be replaced in friction forces. I can open a separate PR to replace those, would that be a good idea?

Reply:
No need to fix all of them (unless you really want to of course) not introducing new ones is enough :)

Reply:
Haha sounds good, I'll do those two since it's quick. Is the benefit to using GeometryToolbox for dot products because it reduces lines of code, and should be faster since its an inline function?

Reply:
Aside from cleanup / expressiveness (i.e. reading a word instead of code), we gain some safety by being able to:
`const su2double a = GeometryToolbox::Something(...)`
Instead of having to allow `a` to be modified.
On the performance side, one of the main sources of inefficiency in SU2 is that the compilers "don't know" that loops over nDim, nVar, nSpecies, are super small, and they often make them inefficient by applying optimizations that are only profitable for large loops.
By encapsulating those loops over nDim we have a single place where we can inform the compilers (via all sorts of compiler-specific directives or in the limit by stating "nDim is 2 or 3") we are not doing this yet.